src/main/java/calculator/shared/PositiveNumber.java

```diff
@@ -0,0 +1,48 @@
+package calculator.shared;
+
+import java.util.Objects;
+
+public class PositiveNumber {
+    private int value;
+
+    private PositiveNumber(int value) {
+        if(value < 0) {
+            throw new IllegalArgumentException("음수는 입력할 수 없습니다.");
+        }
+        this.value = value;
+    }
+
+    public static PositiveNumber of(Integer integer) {
+        return new PositiveNumber(integer);
+    }
+
+    public static PositiveNumber ZERO(){
+        return new PositiveNumber(0);
+    }
```

Comment:
ZERO 라는 값은 호출할 때마다 새로생성하지 않고 재사용할 수 있게 상수로 캐싱해두고 사용해도 좋을것 같아요!

```suggestion
public class PositiveNumber {
    private static final PositiveNumber ZERO = new PositiveNumber(0);
```